Corrected boolean evaluation

itavy opened the pull request:
Hi,

If the value passed to loader is false it will return it initialized as true, since Boolean constructor will return false only for 0, -0, null, false(boolean false), NaN, undefined, or the empty string ("")

here are the steps to reproduce:
```
$ TEST_VAR=false node -e "console.log(require('env-var')('TEST_VAR').asBool())"
true
```

Signed-off-by: Octavian Ionescu <itavyg@gmail.com>

evanshortiss commented on lib/variable.js:
```diff
@@ -79,7 +79,7 @@ module.exports = function environmentVariable (name, value) {
       }
 
-      return Boolean(val);
+      return Boolean('false' === val ? 0 : val);
```
Nice find! I think in my head I was under the impression that _Boolean_ would parse "true" and "false". I'm just thinking that maybe we should also check for "0" and treat it as false too?

evanshortiss replied:
Also, could we make the check not be case sensitive, perhaps like the below code?

```js
if (val && typeof val === 'string') {
  return Boolean('false' === val.toLowerCase() ? 0 : val);
} else {
  return Boolean(val)
}
```

itavy replied:
Hi,

string insensitive is ok; i've seen in shell people using TRUE or FALSE for boolean 

as treating 0 as false i am against type coercion; if someone need boolean value than it shall be boolean not something which can be translated to boolean because of js coercion rules; also it makes the definition of what you expect, more easier to read

looking again over the code, i dont think we need that boilerplate to return a boolean value,
i think it will be easier:

`
return 'false' === val ? false : true;
`

there is allready a check so we can have only 2 values: string 'false' and string 'true'

evanshortiss replied:
@itavy unfortunately this isn't so much a JS issue as an issue of how people on systems declare booleans. Some people use 0 and 1 instead of true and false - although, like you, I prefer to be explicit so the module can simply not support that unless someone needs in the future.

In terms of upper/lower case: some people will use "FALSE" and others will use "false", so it's best to support this. If you can do this then I am happy to merge 👍

itavy commented:
hi,

i updated the method "asBool" to allow the folowing values:
0, 1, true, false and any case writing of true or false

i added "asStrictBoolean" method which allows only true or false values
if it is ok i will update the readme too

evanshortiss commented:
@itavy I see you removed "asStrictBoolean", I didn't mean to remove that, I meant the Array inside the function that wasn't needed. You can add it back if needed. If not, let me know and I will merge.

itavy commented:
hi,

sorry for misunderstanding. i have added back the method and renamed it "asStrictBool" so it will have same semantic as other methods.

btw, to prevent letting unused variables (among many other things) i am using eslint to check the code;

evanshortiss commented:
@itavy nice! Thanks for the PR, merging it now. I need to add an eslint to this repo. Currently it has jshint, but I've moved more to eslint like you.
